cpp/vcpkg.json

```diff
@@ -0,0 +1,40 @@
+{
+  "name": "arrow",
+  "version-string": "4.0.0-SNAPSHOT",
+  "dependencies": [
+    "abseil",
+    {
+      "name": "aws-sdk-cpp",
+      "features": [
+        "config",
+        "cognito-identity",
+        "identity-management",
+        "s3",
+        "sts",
+        "transfer"
+      ]
+    },
+    "benchmark",
```

Comment:
Can we remove this?
It seems that we don't run benchmark with the new job in this pull request.

Reply:
The larger purpose of adding this `vcpkg.json` manifest file is to improve the experience for Windows-based Arrow C++ developers who prefer vcpkg over conda ([ARROW-11336](https://issues.apache.org/jira/browse/ARROW-11336)). We want to give Windows devs two alternative ways to get the same build dependencies: vcpkg or conda. Therefore I want to keep what's here as similar as possible to what's in [ci/conda_env_cpp.yml](https://github.com/apache/arrow/blob/master/ci/conda_env_cpp.yml), which is what we [tell Windows devs to use with conda](https://arrow.apache.org/docs/developers/cpp/windows.html#using-conda-forge-for-build-dependencies). That contains benchmark and gtest, so I think this should too. That way all the latter instructions can be consistent regardless of whether vcpkg or conda was used.

P.S. I can provide the mapping of each conda-forge package to the analogous vcpkg package if you'd like to see that.

Reply:
OK. Then how about adding `-DARROW_BUILD_BENCHMARKS=ON`?